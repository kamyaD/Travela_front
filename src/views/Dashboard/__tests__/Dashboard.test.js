import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import ConnectedDashboard, { Dashboard, mapStateToProps } from '..';

const props = {
  getCurrentUserRole: ['Travel Administrator'],
  history: {
    push: jest.fn()
  },
  departmentTrips: {
    report: [],
    loading: false,
  },
  fetchDepartmentTrips: jest.fn()
};

const mockStore = configureStore();
const store = mockStore({});

const wrapper = mount(
  <Provider store={store}>
    <MemoryRouter>
      <Dashboard {...props} />
    </MemoryRouter>
  </Provider>
);

describe('<Dashboard />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Dashboard {...props} />);
  });


  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('maps the correct state to props', () => {
    const mapper = mapStateToProps(
      {
        user : {getCurrentUserRole: ['Travel Admin']},
        analytics: {departmentTrips: {
          report: [],
          loading: false,
        }}
      });
    expect(mapper.getCurrentUserRole[0]).toEqual('Travel Admin');
  });
});
