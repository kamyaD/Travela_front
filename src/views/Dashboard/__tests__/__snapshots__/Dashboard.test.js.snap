// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders correctly 1`] = `
<div
  id="dashboard"
>
  <FilterContext>
    <ContextConsumer>
      <Component />
    </ContextConsumer>
  </FilterContext>
  <AnalyticsReport
    departmentTrips={
      Object {
        "loading": false,
        "report": Array [],
      }
    }
    exportReadiness={[Function]}
    fetchDepartmentTrips={[Function]}
    fetchReadiness={[Function]}
    readiness={
      Object {
        "isLoading": false,
      }
    }
  />
  <Connect(TravelCalendar) />
</div>
`;
